erpnext/maintenance/doctype/maintenance_visit/maintenance_visit.py

```diff
@@ -16,44 +17,61 @@ def validate_serial_no(self):
 			if d.serial_no and not frappe.db.exists("Serial No", d.serial_no):
 				frappe.throw(_("Serial No {0} does not exist").format(d.serial_no))
 
+	def validate_maintenance_date(self):
+		if self.maintenance_type == "Scheduled" and self.maintenance_schedule_detail:
+			item_ref = frappe.db.get_value('Maintenance Schedule Detail', self.maintenance_schedule_detail, 'item_reference')
+			start_date, end_date = frappe.db.get_value('Maintenance Schedule Item', item_ref, ['start_date', 'end_date'])
```

Comment:
Pre-existing maintenance schedule detail table does not have `item_reference`. 
`item_ref` will be `None`
And so **line 23** breaks as you cannot unpack a Nonetype object, as its not an iterable.

Check if `item_ref` is available and then proceed